File: backend/app/schemas/payment.py
from pydantic import BaseModel
from datetime import datetime
from typing import Optional
from app.models.payment import MilestoneType, MilestoneStatus, MilestoneRatio


class PaymentUpdate(BaseModel):
    paid_amount: Optional[float] = None
    total_amount: Optional[float] = None
    mratio: Optional[MilestoneRatio] = None
    status: Optional[MilestoneStatus] = None
    paid_at: Optional[datetime] = None
    due_date: Optional[datetime] = None


class PaymentResponse(BaseModel):
    payment_id: int
    sale_id: int
    milestone: MilestoneType

    total_amount: Optional[float] = None
    paid_amount: Optional[float] = None
    mratio: Optional[MilestoneRatio] = None

    status: MilestoneStatus
    paid_at: Optional[datetime] = None
    due_date: Optional[datetime] = None

    class Config:
        from_attributes = True

File: frontend/src/types/payment.ts
// types/payment.ts
export type PaymentMilestone =
  | "TOKEN"
  | "ATS"
  | "SUPERSTRUCTURE"
  | "PROPERTY_ID"
  | "REGISTRY"
  | "POSSESSION"

export type PaymentStatus = "PENDING" | "DONE"

export type Payment = {
  payment_id: number
  sale_id: number
  milestone: PaymentMilestone
  amount?: number
  total_amount?: number | null
  paid_amount?: number | null
  mratio?: string | null
  status: PaymentStatus
  paid_at?: string | null
  due_date?: string | null
}
